include/librealsense2/hpp/rs_processing.hpp

```diff
@@ -432,7 +439,8 @@ namespace rs2
         {
             (*_block)(frame);
             rs2::frame f;
-            _queue.poll_for_frame(&f);
+            if (!_queue.poll_for_frame(&f))
```

Comment:
The four lines code are repeated for every processing block - can be refactored into a base class. 
Introducing "base logic" class may also provide a scope to address frameset decomposition-recreation in a unified form.